src/NuGet.Core/NuGet.Commands/PackCommandRunner.cs

```diff
@@ -516,8 +531,14 @@ private PackageArchiveReader BuildFromProjectFile(string path)
                 factory.GetProjectProperties()[property.Key] = property.Value;
             }
 
+            NuGetVersion version = null;
+            if (_packArgs.Version != null)
+            {
+                version = new NuGetVersion(_packArgs.Version);
+            }
+
             // Create a builder for the main package as well as the sources/symbols package
-            PackageBuilder mainPackageBuilder = factory.CreateBuilder(_packArgs.BasePath);
+            PackageBuilder mainPackageBuilder = factory.CreateBuilder(_packArgs.BasePath, version, _packArgs.Suffix, buildIfNeeded: true);
```

Comment:
I started doing something more like that, but you can pass in just a suffix and no version to pack while having a version in project.json.  It needs to know that the version passed in was null but the suffix wasn't null.  So it seemed that we needed to pass around both.